dbms/src/DataTypes/DataTypesDecimal.h

```diff
@@ -332,12 +332,12 @@ convertToDecimal(const typename FromDataType::FieldType & value, UInt32 scale)
         {
             static constexpr __int128 min_int128 = __int128(0x8000000000000000ll) << 64;
             static constexpr __int128 max_int128 = (__int128(0x7fffffffffffffffll) << 64) + 0xffffffffffffffffll;
-            if (out < min_int128 || out > max_int128)
+            if (! (out > min_int128 && out < max_int128))
```

Comment:
We'd better manually cast int to float here
```
if (out <= static_cast<ToNativeType>(min_int128) || out >= static_cast<ToNativeType>(max_int128))
```

Reply:
do we need? i see `<ToNativeType>` is always int128 here
https://github.com/yandex/ClickHouse/blob/21a48e4f6c549a313e0689fb6869fea90bd2681a/dbms/src/DataTypes/DataTypesDecimal.h#L331

Reply:
My bad. I mean not ToNativeTipe but FromFieldType. We have to compare them as Float64 not as Int128 here. So we force overflow for all the values that have same conversion to Float64/Float32 as int128_max / int128_min.
You make also look at AccurateComparison.h content.